src/components/v5/common/CompletedAction/partials/SetUserRoles/SetUserRoles.tsx

```diff
@@ -45,11 +50,22 @@ interface Props {
 
 const transformActionRolesToColonyRoles = (
   roles: ColonyActionRoles | null | undefined,
+  historicRoles: GetColonyHistoricRoleRolesQuery['getColonyHistoricRole'],
 ): ColonyRole[] => {
   if (!roles) return [];
-  const roleKeys = Object.keys(roles);
+
+  const combinedRoles = { ...historicRoles };
+
+  for (const [key, value] of Object.entries(roles)) {
+    if (value !== null) {
+      combinedRoles[key] = value;
+    }
+  }
+
+  const roleKeys = Object.keys(combinedRoles);
```

Comment:
I believe (please double check though) historic roles entry is a snapshot of the entire permissions state so we should be able to use it without merging with with `roles` - which would make that first argument redundant.